ng-deploy/src/builders/actions/deploy.ts

```diff
@@ -1,8 +1,40 @@
-import { execSync, spawnSync } from "child_process";
+import { BuilderContext } from '@angular-devkit/architect/src/index2';
+import { virtualFs, experimental, normalize } from '@angular-devkit/core';
+import { Stats } from '@angular-devkit/core/src/virtual-fs/host';
+import { CoreSchemaRegistry, transforms } from '@angular-devkit/core/src/json/schema';
+import { join } from 'path';
+import { FirebaseTools } from '../../ng-deploy/types';
 
-export default function deploy() {
-    const bin = execSync('(cd ' + __dirname + ' && npm bin)').toString().trim();
-    const firebaseBin = bin + '/firebase';
-    spawnSync(firebaseBin, ['login'], {stdio: 'inherit'});
-    spawnSync(firebaseBin, ['deploy', '--only', 'hosting'], {stdio: 'inherit'});
+export default async function deploy(firebaseTools: FirebaseTools, context: BuilderContext, host: virtualFs.Host<Stats>,) {
+  try {
+    await firebaseTools.list();
+  } catch (e) {
+    context.logger.warn("🚨 You're not logged into Firebase. Logging you in...");
+    await firebaseTools.login();
+  }
+  if (!context.target) {
+    throw new Error('Cannot execute the build target');
+  }
+
+  context.logger.info(`📦 Building "${context.target.project}"`);
+
+  const run = await context.scheduleTarget({ target: 'build', project: context.target.project, configuration: 'production' });
+  await run.result;
+  if (!context.target) {
+    throw new Error('Cannot execute the build target');
+  }
+
+  const registry = new CoreSchemaRegistry();
```

Comment:
I don't understand this one, we create an instance, mutate in, and then don't do anything with it?

Reply:
Just redundant code. Dropped it.